### ISSUE TITLE
cant edit networked files (ssh / sftp)

### ISSUE THREAD

**Issue by mf0hka** [geany/geany#893]
I believe formerly in 1.21 or around there, editing of files was allowed. Im not sure what happened, but this seems to be the case no longer. Pluma, does this operation just fine, as does libre office, and some other file editors. When I try to explicitly run geany to open a file across the network, geany will open but the file will not. The types of connections
Here it is working in pluma, which is what I would like to see fixed in geany. Pluma is a fine text editor, but I like Geany far more. 
![s](https://cloud.githubusercontent.com/assets/6516504/12702345/7163582a-c7f4-11e5-8a77-bbb0509a27d7.png)
 are for me, ssh, and sftp.

**Comment by codebrainz** [comment]
The following information would be useful:

- What version of Geany are you using?
- What operation system and version are you using?
- How have you mounted the remote file systems?
- What happens when you try to open the file specifically? (ex. error messages, debug messages, etc)
- Have you seen sf.net [bug #1043](http://sourceforge.net/p/geany/bugs/1043/)?

**Comment by elextr** [comment]
The ability of Geany to open files has not changed.  What has most likely
changed is that your system is not auto-mounting them locally any more.

**Comment by derba** [comment]
I guess one should not implement a file system driver (ssh, sftp, ftp, nfs, smb etc) in a text editor. That is an OS or desktop environment functionality.

BTW, in File / Open I can write ssh://... and it works fine (gvfs does it, I think).

**Comment by elextr** [comment]
> I guess one should not implement a file system driver (ssh, sftp, ftp, nfs, smb etc) in a text editor. That is an OS or desktop environment functionality.

Yes, note that pluma is really a part of the DE and is tightly integrated and uses the Gnome Virtual File System to read and write.  But being portable Geany is not so tightly integrated.

**Comment by elextr** [comment]
@derba I just noticed that the add-ons plugin has an "Add Open URI item in edit menu", not sure if that will do anything for you?

**Comment by arigit** [comment]
This seems related to: 

https://github.com/geany/geany/issues/1437

Geany 1.30.1 on GNOME 3.22.3 (using it on Fedora 25).

Up to this last update, editing files on sftp mounts done with nautilus was working perfectly. For years. 

Right now, opening a file mounts via sftp in nautilus: geany shows the file as if it had no content at all, and if "save" is invoked it will overwrite the original file, all the contents are lost, people can easily lose code due to this problem - even worse: this can haven with auto-save as well.
Please take a look
